homeassistant/components/tellduslive/config_flow.py

```diff
@@ -0,0 +1,121 @@
+"""Config flow for Tellduslive."""
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.util.json import load_json
+
+from .const import (APPLICATION_NAME, CLOUD_NAME, DOMAIN, KEY_HOST,
+                    NOT_SO_PRIVATE_KEY, PUBLIC_KEY, TELLDUS_CONFIG_FILE)
+
+KEY_TOKEN = 'token'
+KEY_TOKEN_SECRET = 'token_secret'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register('tellduslive')
+class FlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Init config flow."""
+        self._hosts = [CLOUD_NAME]
+        self._host = None
+        self._session = None
+
+    async def async_step_user(self, user_input=None):
+        """Let user select host or cloud."""
+        from tellduslive import Session
+        if user_input is not None or len(self._hosts) == 1:
+            if user_input is not None and user_input[KEY_HOST] != CLOUD_NAME:
+                self._host = user_input[KEY_HOST]
+            self._session = Session(
+                public_key=PUBLIC_KEY,
+                private_key=NOT_SO_PRIVATE_KEY,
+                host=self._host,
+                application=APPLICATION_NAME,
+            )
+            return await self.async_step_auth()
+
+        return self.async_show_form(
+            step_id='user',
+            data_schema=vol.Schema({
+                vol.Required(KEY_HOST):
+                vol.In(list(self._hosts))
+            }))
+
+    async def async_step_auth(self, user_input=None):
+        """Handle the submitted configuration."""
+        if user_input is not None and self._session.authorize():
+            host = self._host or CLOUD_NAME
+            if self._host:
+                data = {KEY_HOST: host, KEY_TOKEN: self._session.access_token}
+            else:
+                data = {
+                    KEY_TOKEN: self._session.access_token,
+                    KEY_TOKEN_SECRET: self._session.access_token_secret
+                }
+            return self.async_create_entry(title=host, data=data)
+
+        try:
+            auth_url = self._session.authorize_url
+        except Exception:  # pylint: disable=broad-except
```

Comment:
Why are we expecting unknown errors here? What happens in `self._session.authorize_url`?

Reply:
There is actually IO going on in that method, I'll revert the change..